[docs][screen-orientation] Document iOS 27 resizability (#48174)
expo/expo: docs/pages/versions/unversioned/sdk/screen-orientation.mdx

> Web has [limited support](https://caniuse.com/#feat=deviceorientation).

> ## Orientation locks on iOS 27 and later

Reviewer: ✏️ **style** `mdx-components-sdk.md#platformtags`

This section is iOS-only on a page tagged for Android, iOS, and web, so badge the heading: `## Orientation locks on iOS 27 and later&ensp;<PlatformTags platforms={['ios']} />`. That component needs `import { PlatformTags } from '~/ui/components/Tag/PlatformTags';` added with the other imports, the same way `background-fetch.mdx` and `imagepicker.mdx` mark their iOS-only sections.

<!-- docs-review -->

Starting in iOS 27, iPhone apps can be resized, for example in iPhone Mirroring on macOS or when running on an iPad. While an app is resizable, the system treats a supported orientation as a preference rather than a requirement. This has two consequences:

- `lockAsync` may have no effect. The system can refuse the request, and your app stays resizable.
- `getOrientationAsync` may not describe the shape of the window your app is drawing into. iPhone Mirroring always reports a portrait orientation regardless of the window's aspect ratio.

> Design your screens to adapt to the space available to them instead of relying on a fixed orientation. In debug builds, this library logs a warning when the system refuses an orientation lock, so you can find the places where your app depends on behavior the platform no longer guarantees.

Reviewer: ✏️ **style** `prose.md#core-principles`

The second sentence runs 32 words and carries three ideas. Split the warning off from what the reader does with it.

```suggestion
Design your screens to adapt to the space available to them instead of relying on a fixed orientation. In debug builds, this library logs a warning when the system refuses an orientation lock. Use those warnings to find the screens that still depend on a lock the platform no longer guarantees.
```

<!-- docs-review -->

## Installation

<APIInstallSection />

### Warning

> Apple added support for _split view_ mode to iPads in iOS 9. This changed how the screen orientation is handled by the system. To put the matter shortly, for iOS, your iPad is always in landscape mode unless you open two applications side by side. To be able to lock screen orientation using this module you will need to disable support for this feature by setting `ios.requireFullScreen` to `true` in your app config.

Reviewer: ```suggestion
Apple added support for _split view_ mode to iPads in iOS 9. This changed how the screen orientation is handled by the system. To put the matter shortly, for iOS, your iPad is always in landscape mode unless you open two applications side by side. To be able to lock screen orientation using this module you will need to disable support for this feature by setting [`ios.requireFullScreen`](/versions/latest/config/app/#requirefullscreen) to `true` in your [app config](/workflow/configuration/).
```

Let's add internal links here.

> On iOS 27 and later, `requireFullScreen` no longer opts your app out of being resized, so it is no longer a reliable way to lock the screen orientation. See [Orientation locks on iOS 27 and later](#orientation-locks-on-ios-27-and-later). For more information about the _split view_ mode, check out [the official Apple documentation](https://support.apple.com/en-us/HT207582).

Reviewer: 💡 **nit** `prose.md#core-principles`

"no longer opts" and "no longer a reliable way" land in one 28-word sentence, so tighten the second half.

```suggestion
On iOS 27 and later, `requireFullScreen` no longer opts your app out of being resized, so it cannot reliably lock the screen orientation. See [Orientation locks on iOS 27 and later](#orientation-locks-on-ios-27-and-later). For more information about the _split view_ mode, check out [the official Apple documentation](https://support.apple.com/en-us/HT207582).
```

<!-- docs-review -->
